src/Toolchain/ToolchainProvider.ts

```diff
@@ -28,14 +28,15 @@ export class ToolchainNode extends vscode.TreeItem {
       public readonly label: string,
       public readonly collapsibleState: vscode.TreeItemCollapsibleState,
       public readonly type: NodeType,
+      public readonly backend?: string,
```

Comment:
Q. Any reason to assign `backend` as `string | undefined`? If you assign it as just the only `string`, we don't have to care about `backend` is undefined in ts.

Reply:
The `backend` is optional parameter so it can either have a value based on the type defined or its value can be `undefined`. However, when I use `backend: string | undefined` code, this value must be inputted. But `NodeType.backend` does not use `backend` string. So I use it as optional parameter.

Reply:
I understand your point.